src/pal/src/misc/time.cpp

```diff
@@ -270,6 +278,20 @@ GetTickCount64()
         }
         retval = (ts.tv_sec * tccSecondsToMillieSeconds)+(ts.tv_nsec / tccMillieSecondsToNanoSeconds);
     }
+#elif HAVE_MACH_ABSOLUTE_TIME
+    {
+        // use denom == 0 to indicate that s_TimebaseInfo is uninitialised.
+        if (s_TimebaseInfo.denom == 0)
```

Comment:
I think there is a potential race if multiple threads are executing this code in parallel. One of the threads can return bogus result.